relayer/provider/cosmos/provider.go

```diff
@@ -1153,55 +1153,53 @@ func (cc *CosmosProvider) relayPacketsFromResultTx(ctx context.Context, src, dst
 		timeoutPackets []provider.RelayPacket
 	)
 
-	rp := &relayMsgRecvPacket{pass: false}
+EventLoop:
 	for _, event := range resp.Events {
-		rp.pass = false
+		rp := &relayMsgRecvPacket{}
 
 		if event.EventType != spTag {
 			continue
 		}
 
-		switch event.AttributeKey {
-		case srcChanTag:
-			if event.AttributeValue != srcChanId {
-				rp.pass = true
-				continue
-			}
-		case dstChanTag:
-			if event.AttributeValue != dstChanId {
-				rp.pass = true
-				continue
-			}
-		case srcPortTag:
-			if event.AttributeValue != srcPortId {
-				rp.pass = true
-				continue
+		for attributeKey, attributeValue := range event.Attributes {
+			switch attributeKey {
+			case srcChanTag:
+				if attributeValue != srcChanId {
+					continue EventLoop
+				}
+			case dstChanTag:
+				if attributeValue != dstChanId {
+					continue EventLoop
+				}
+			case srcPortTag:
+				if attributeValue != srcPortId {
+					continue EventLoop
+				}
+			case dstPortTag:
+				if attributeValue != dstPortId {
+					continue EventLoop
+				}
+			case dataTag:
+				rp.packetData = []byte(attributeValue)
+			case toHeightTag:
+				timeout, err := clienttypes.ParseHeight(attributeValue)
+				if err != nil {
+					continue EventLoop
```

Comment:
I changed these from returning the error to continuing on to the next event. I figured we would want to continue attempting to parse other events in the tx (and hold onto potentially prior successfully parsed events) even if a single event fails to parse.

Reply:
Log the error there? Swallowing errors in Go has resulted in great pain in my past.

Reply:
good call. added